task_4/task_4.py

```diff
@@ -0,0 +1,65 @@
+# Download links.parquet from https://drive.google.com/file/d/1ym_RbsjN41cwXZLeB3NibN7h7Vbz1AgP/view?usp=sharing
+# Download and save the first 10,000 images from the links in the parquet file.
+# - Monitor CPU and network usage during download.
+
+# References:
+# pyarrow package
+# requests package
+
+import pyarrow.parquet as pq
+import requests
+import os
+
+# extracting all info in links.parquet and storing it as a pandas table
+table = pq.read_table("links.parquet")
+df = table.to_pandas()

```

Comment:
Don't read full table and convert to pandas. It will take more time and space. `pq.read_table` has a `columns` parameter.

Reply:
ok I will update that